Library/Homebrew/cask/audit.rb

```diff
@@ -611,7 +611,7 @@ def audit_min_os
 
       return if min_os.nil? || min_os <= HOMEBREW_MACOS_OLDEST_ALLOWED
 
-      cask_min_os = if cask.on_system_blocks_exist?
+      cask_min_os = if cask.on_system_blocks_exist? && cask.on_system_block_min_os
```

Comment:
```suggestion
      cask_min_os = if cask.on_system_blocks_exist? && cask.on_system_block_min_os.present?
```

I think this matches the convention we use? It's been a little since I've been deeply involved though so maybe that's changed